language/evm/move-to-yul/tests/Arithm.exp

```diff
@@ -0,0 +1,447 @@
+                $result := $t3
+            }
+
+            function A2_M_div_by_zero_u64(x, y) -> $result {
+                let $t2
+                /// @src 2:2685:2686
+                $t2 := $Div(x, y)
+                /// @src 2:2683:2688
+                $result := $t2
+            }
+
+            function A2_M_div_by_zero_u64_incorrect(x, y) -> $result {
+                let $t2
+                /// @src 2:2533:2534
+                $t2 := $Div(x, y)
+                /// @src 2:2531:2536
+                $result := $t2
+            }
+
+            function A2_M_f(x) -> $result {
+                let $t1, $t2
+                $t1 := 1
+                /// @src 2:1543:1544
+                $t2 := $AddU64(x, $t1)
+                /// @src 2:1542:1545
+                $result := $t2
+            }
+
+            function A2_M_g(x) -> $result {
+                let $t1, $t2
+                $t1 := 2
+                /// @src 2:1689:1690
+                $t2 := $AddU64(x, $t1)
+                /// @src 2:1688:1691
+                $result := $t2
+            }
+
+            function A2_M_h(b) -> $result {
+                let tmp_$1, tmp_$2, x, y, $t5, $t6, $t7, $t8, $t9, $t10, $t11, $t12
+                let $block := 4
+                for {} true {} {
+                    switch $block
+                    case 2 {
+                        y := A2_M_f($t5)
+                        /// @src 2:1885:1891
+                        $block := 5
+                    }
+                    case 3 {
+                        y := A2_M_g($t5)
+                        /// @src 2:1917:1918
+                        $block := 5
+                    }
+                    case 4 {
+                        $t5 := 3
+                        /// @src 2:1878:1903
+                        if b { $block := 2 } else { $block := 3 }
+                    }
+                    case 5 {
+                        /// @src 2:1917:1928
+                        if b { $block := 6 } else { $block := 7 }
+                    }
+                    case 6 {
+                        $t6 := 4
+                        /// @src 2:1924:1926
+                        tmp_$1 := $Neq(y, $t6)
+                        /// @src 2:1917:1928
+                        $block := 8
+                    }
+                    case 7 {
+                        $t7 := false
+                        /// @src 2:1917:1928
+                        tmp_$1 := $t7
+                        /// @src 2:1917:1928
+                        $block := 8
+                    }
+                    case 8 {
+                        /// @src 2:1913:1937
+                        if tmp_$1 { $block := 9 } else { $block := 10 }
+                    }
+                    case 9 {
+                        $t8 := 4
+                        $Abort($t8)
+                    }
+                    case 10 {
+                        /// @src 2:1951:1952
+                        $t9 := $LogicalNot(b)
```

Comment:
The Solidity compiler complains here: "std::exception::what: Variable b is 1 slot(s) too deep inside the stack". Maybe there are too many local variables on the stack while DUP or SWAP operations can only work until 16

Reply:
This is unfortunately a serious restriction of Yul. The only workaround seems to put variables into an array on the heap once there are more than a dozen or so.

https://github.com/ethereum/solidity/issues/6928